jdaviz/configs/default/plugins/model_fitting/model_fitting.py

```diff
@@ -581,6 +586,50 @@ def set_model_component(self, model_component_label, parameter, value=None, fixe
 
         return parameter
 
+    def vue_reestimate_model_parameters(self, model_component_label=None, **kwargs):
+        self.reestimate_model_parameters(model_component_label=model_component_label)
+
+    def reestimate_model_parameters(self, model_component_label=None):
+        """
+        Re-estimate all free parameters in a given model component given the currently selected
+        data and subset selections.
+
+        Parameters
+        ----------
+        model_component_label : str or None.
+            The label given to the existing model component.  If None, will iterate over all model
+            components.
+        """
+        if model_component_label is None:
+            return [self.reestimate_model_parameters(model_comp["id"])
+                    for model_comp in self.component_models]
+
+        try:
+            model_index, model_comp = [(i, x) for i, x in enumerate(self.component_models)
+                                       if x["id"] == model_component_label][0]
+        except IndexError:
+            raise ValueError(f"'{model_component_label}' is not a label of an existing model component")  # noqa
+
+        # store user-fixed parameters so we can revert after re-initializing
+        fixed_params = {p['name']: p for p in model_comp['parameters'] if p['fixed']}
+
+        new_model = self._initialize_model_component(model_comp['model_type'],
+                                                     model_component_label,
+                                                     poly_order=model_comp['model_kwargs'].get('degree', None))  # noqa
+
+        # revert fixed parameters to user-value
+        new_model['parameters'] = [fixed_params.get(p['name'], p) for p in new_model['parameters']]
+
+        # reset traitlet so vue picks up on changes by adding (and then removing) an empty entry,
+        # this avoids resetting the open state of the accordion in the UI.
```

Comment:
🤯

Reply:
I'd love to know of a better way to do this... or requesting/implementing `traitlet.force_update()` upstream...

Reply:
Maybe @mariobuikhuizen can advise but this shouldn't block merge. Just amazed you figured out this is how you make it do what you want. To me, it is like eating pizza with chopsticks. 😸

Reply:
Traitlets can't discover in-place changes to a list or dict, so you always need to use methods that produce a new instance. You could consider using:
```python
self.component_models = self.component_models[:model_index] + [new_model] + self.component_models[model_index + 1:]
```

Or you could use `self.send_state("component_models")`, which is similar to the envisioned `.force_update()` function.

Reply:
I didn't know about `send_state`, that seems to do the trick and reads more clearly to me than always rebuilding from scratch, thanks!